guides/v2.1/contributor-guide/contributing_dod.md

```diff
@@ -163,6 +163,20 @@ A task, when complete by the author, should be reviewed by another Magento commu
 The reviewer should check whether the task meets the original acceptance criteria and verify that there are no code defects and that other points of this definition of done are met:
 
 * There are no unauthorized backward-incompatible functional changes
+* All backward-incompatible changes on code level are approved by an architect. The following cases currently cause SVC failures but do NOT require architect approval:
+  * Removal of private constant
+  * Override of the method defined in parent class
+  * Adding optional parameter to the constructor, except for the classes explicitly meant for extension, including the following ones:
+    * `\Magento\Framework\Model\AbstractExtensibleModel`
+    * `\Magento\Framework\Api\AbstractExtensibleObject`
+    * `\Magento\Framework\Api\AbstractSimpleObject`
+    * `\Magento\Framework\Model\AbstractModel`
+    * `\Magento\Framework\App\Action\Action`
+    * `\Magento\Backend\App\Action`
+    * `\Magento\Backend\App\AbstractAction`
+    * `\Magento\Framework\App\Action\AbstractAction`
+    * `\Magento\Framework\View\Element\AbstractBlock`
+    * `\Magento\Framework\View\Element\Template`
```

Comment:
Would it makes sense to add `\Magento\Framework\Data\Collection\AbstractDb`, `\Magento\Eav\Model\Entity\Collection\AbstractCollection`, `\Magento\Eav\Model\Entity\Collection\VersionControl\AbstractCollection` to the list?

Reply:
Not sure about `\Magento\Eav\Model\Entity\Collection\AbstractCollection` and `\Magento\Eav\Model\Entity\Collection\VersionControl\AbstractCollection` as they are module specific classes.

Reply:
You extend these if your entity has EAV attributes.